src/logic/tokens/store/actions/fetchTokens.js

```diff
@@ -37,25 +39,119 @@ const createERC721TokenContract = async () => {
   return erc721Token
 }
 
+const OnlyBalanceToken = {
+  contractName: 'OnlyBalanceToken',
+  abi: [
+    {
+      constant: true,
+      inputs: [
+        {
+          name: 'owner',
+          type: 'address',
+        },
+      ],
+      name: 'balanceOf',
+      outputs: [
+        {
+          name: '',
+          type: 'uint256',
+        },
+      ],
+      payable: false,
+      stateMutability: 'view',
+      type: 'function',
+    },
+    {
+      constant: true,
+      inputs: [
+        {
+          name: 'owner',
+          type: 'address',
+        },
+      ],
+      name: 'balances',
+      outputs: [
+        {
+          name: '',
+          type: 'uint256',
+        },
+      ],
+      payable: false,
+      stateMutability: 'view',
+      type: 'function',
+    },
+  ],
+}
+
+// For the `batchRequest` of balances, we're just using the `balanceOf` method call.
+// So having a simple ABI only with `balanceOf` prevents errors
+// when instantiating non-standard ERC-20 Tokens.
+const createOnlyBalanceToken = () => {
+  const web3 = getWeb3()
+  const contract = new web3.eth.Contract(OnlyBalanceToken.abi)
+  return contract
+}
+
 export const getHumanFriendlyToken = ensureOnce(createHumanFriendlyTokenContract)
 
 export const getStandardTokenContract = ensureOnce(createStandardTokenContract)
 
 export const getERC721TokenContract = ensureOnce(createERC721TokenContract)
 
+export const getOnlyBalanceToken = ensureOnce(createOnlyBalanceToken)
+
 export const containsMethodByHash = async (contractAddress: string, methodHash: string) => {
   const web3 = getWeb3()
   const byteCode = await web3.eth.getCode(contractAddress)
 
   return byteCode.indexOf(methodHash.replace('0x', '')) !== -1
 }
 
-export const fetchTokens = () => async (dispatch: ReduxDispatch<GlobalState>) => {
+export const getTokenInfos = async (tokenAddress: string) => {
+  if (!tokenAddress) {
+    return null
+  }
+  const { tokens } = store.getState()
+  let tokenInstance = tokens.get(tokenAddress)
+  // If the token is inside the store we return the store token
+  if (tokenInstance) {
+    return tokenInstance
+  }
+  // Otherwise we fetch it, save it to the store and return it
+  const tokenContract = await getHumanFriendlyToken()
+  tokenInstance = await tokenContract.at(tokenAddress)
+  const [tokenSymbol, tokenDecimals, name] = await Promise.all([
+    tokenInstance.symbol(),
+    tokenInstance.decimals(),
+    tokenInstance.name(),
+  ])
+  tokenInstance.symbol = tokenSymbol
+  tokenInstance.decimals = tokenDecimals
```

Comment:
I think we don't need to do any assignments

Reply:
Yep because functions that uses getTokenInfos expects to have the object and not a promise

Reply:
I don't think passing contract instance with reassigned properties is a good/safe idea. I'd would to the following: 

```
const [symbol, decimals, name] = await Promise.all([
    tokenInstance.symbol(),
    tokenInstance.decimals(),
    tokenInstance.name(),
  ])

return { symbol, decimals, name }
```

Reply:
you could also return `savedToken`